ams/ams-server/src/main/java/com/netease/arctic/ams/server/ArcticMetaStore.java

```diff
@@ -321,12 +322,14 @@ private static void syncAndExpiredFileInfoCache(final TServer server) {
             Thread.sleep(5 * 60 * 1000);
           } catch (InterruptedException e) {
             LOG.warn("sync and expired file info cache thread was interrupted: " + e.getMessage());
+            return;
           }
         }
         try {
           Thread.sleep(60 * 1000);
         } catch (InterruptedException e) {
           LOG.warn("sync and expired file info cache thread was interrupted: " + e.getMessage());
+          return;
```

Comment:
while(true) is risky, if the thread is doing `task.doTask()` when interrupted, it can't exit.

Reply:
drop while(true), only retain while(server.isServering)